src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.cpp

```diff
@@ -0,0 +1,506 @@
+{
+    ChipLogDetail(Zcl, "PushAV: InvokeCommand");
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Commands::AllocatePushTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Allocating Push Transport");
+
+        HandleCommand<Commands::AllocatePushTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleAllocatePushTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::DeallocatePushTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Deallocating Push Transport");
+
+        HandleCommand<Commands::DeallocatePushTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleDeallocatePushTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::ModifyPushTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Modifying Push Transport");
+
+        HandleCommand<Commands::ModifyPushTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleModifyPushTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::SetTransportStatus::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Setting Push Transport Status");
+
+        HandleCommand<Commands::SetTransportStatus::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleSetTransportStatus(ctx, commandData); });
+
+        break;
+
+    case Commands::ManuallyTriggerTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Manually Triggered Push Transport");
+
+        HandleCommand<Commands::ManuallyTriggerTransport::DecodableType>(
+            handlerContext,
+            [this](HandlerContext & ctx, const auto & commandData) { HandleManuallyTriggerTransport(ctx, commandData); });
+
+        break;
+
+    case Commands::FindTransport::Id:
+        ChipLogDetail(Zcl, "PushAVStreamTransport: Finding Push Transport");
+
+        HandleCommand<Commands::FindTransport::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleFindTransport(ctx, commandData); });
+
+        break;
+    default:
+        // Mark unrecognized command as UnsupportedCommand
+        handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath, Status::UnsupportedCommand);
+        break;
+    }
+}
+
+TransportConfigurationStruct * PushAvStreamTransportServer::FindStreamTransportConnection(const uint16_t connectionID)
+{
+    for (auto & transportConnection : mCurrentConnections)
+    {
+        if (transportConnection.connectionID == connectionID)
+            return &transportConnection;
+    }
+    return nullptr;
+}
+
+uint16_t PushAvStreamTransportServer::GenerateConnectionID()
+{
+    static uint16_t lastAssignedConnectionID = 0;
+    uint16_t nextConnectionID;
+
```

Comment:
How is this working?
nextConnectionID is always going to be either 0 or 1, lastAssigned never changes inside the loop. If you have connections already for those connection IDs, isn't this going to loop forever?